tracing-appender/src/rolling.rs

```diff
@@ -253,21 +227,37 @@ impl RollingFileAppender {
 
 impl io::Write for RollingFileAppender {
     fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+        let should_rollover = |now: OffsetDateTime| -> bool {
+            if let Some(current_time) = self.state.should_rollover_by_date(now) {
+                let _did_cas = self.state.advance_date(now, current_time);
+                debug_assert!(_did_cas, "if we have &mut access to the appender, no other thread can have advanced the timestamp...");
```

Comment:
question: I see you added a `_` in front of `did_cas`. Was it a problem to keep `did_cas` as-is? The `_` is suppressing "unused" warnings, so unless we get a spurious warning I would prefer to not use this prefix.

Reply:
I left it as it was before, but you are right, we don't need to underscore it.

Reply:
Oh, I forgot I was the one who changed it. My bad! Feel free to revert or keep my change. :slightly_smiling_face: